lib/config/presets/index.ts

```diff
@@ -222,15 +227,33 @@ export async function getPreset(
   const { presetSource, repo, presetPath, presetName, tag, params } =
     parsePreset(preset);
   const cacheKey = `preset:${preset}`;
-  let presetConfig = memCache.get<Preset | null | undefined>(cacheKey);
+  const presetCache = GlobalConfig.get('presetCache', false);
+
+  let presetConfig: Preset | null | undefined;
+
+  if (presetCache) {
+    presetConfig = await packageCache.get(presetCacheNamespace, cacheKey);
+  } else {
+    presetConfig = memCache.get(cacheKey);
+  }
```

Comment:
Reverse this order so that the memcache is consulted first. Also ensure that the value is written to memCache if it's looked up from packageCache. Doing so will increase efficiency if the same preset is loaded more than once per repo.

Reply:
Good idea. However what's been described seems to have already been implemented in [packageCache.get](https://github.com/renovatebot/renovate/blob/main/lib/util/cache/package/index.ts#L20-L26) & [packageCache.set](https://github.com/renovatebot/renovate/blob/main/lib/util/cache/package/index.ts#L46-L53). Would you still like the behaviour to be replicated here?

Reply:
I'm not sure I understand. I'll describe in more details what I was suggesting:
- Always check the memcache first, because it will be quicker than presetCache
- If memcache returns nothing, check presetCache
- If presetCache returns something, add it to the memory cache (unless we have logic which does this already?)

Reply:
> unless we have logic which does this already

That's what I meant. please see [packageCache.get](https://github.com/renovatebot/renovate/blob/main/lib/util/cache/package/index.ts#L20-L26) and [packageCache.set](https://github.com/renovatebot/renovate/blob/main/lib/util/cache/package/index.ts#L46-L53)